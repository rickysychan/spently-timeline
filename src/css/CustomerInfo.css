.customer-info{
    font-family: Karla, Helvetica, Arial, sans-serif;
    transform: translateY(1em);
    transition: all 0.3s ease-in-out 0s, visibility 0s linear 0.3s, linear 0.01s;
}

.customerInfo-image{
    height: 150px;
    width: 150px;
    float: left;
    margin-right: 20px;
    margin-left: 15px;
}

.customerInfo-details{
    margin-left: 15px;
    font-size: 1.2rem;
    margin: 15px;
}

.customerInfo-btn{
    text-align: center;
    color: white;
    background: linear-gradient(#ff5b57, #E4504D);
    border: 1px solid #E4504D;
    box-sizing: border-box;
    padding: 8px 10px;
    border-radius: 3px;
    font-size: 1rem;
    letter-spacing: -0.02em;
    font-weight: 400;
    font-family: "Karla",Helvetica,Arial,sans-serif;
    text-decoration: none;
    cursor: pointer;
    outline: none;
    margin:0 auto;
    display:block;
    margin-bottom: 25px;
}

#customerInfo-name{
    margin-top: 0px;    
}