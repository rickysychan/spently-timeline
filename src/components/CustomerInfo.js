import React from 'react'
import PropTypes from 'prop-types'
import '../css/CustomerInfo.css'

const customerInfo = props => {
    if(!props.selectedCustomer){
        return(
            <div className='customerInfo-no-customer'></div>
        )
    }
        else {
            return(
                <div className='customer-info'>
                    <img className='customerInfo-image' src={props.selectedCustomer.picture} alt='customer'/>
                    <h6 className='customerInfo-details'>Full Name: {props.selectedCustomer.name.first + " " + props.selectedCustomer.name.last}</h6>
                    <h6 className='customerInfo-details'>Balance: {props.selectedCustomer.balance}</h6>
                    <h6 className='customerInfo-details'>Active: {props.selectedCustomer.isActive.toString()}</h6>
                    <h6 className='customerInfo-details'>Age: {props.selectedCustomer.age}</h6>
                    <h6 className='customerInfo-details'>About: {props.selectedCustomer.about}</h6>
                    
                    <h6 className='customerInfo-details'>Address: {props.selectedCustomer.address}</h6>
                    <h6 className='customerInfo-details'>Friends: {props.selectedCustomer.friends.map( (friend, index) => 
                            <span key={index}> {friend.name + ", "} </span>
                             )}</h6>
                    <h6 className='customerInfo-details'>Tags: {props.selectedCustomer.tags.map( (tag, index)  => 
                            <span key={index}>{tag + ', '}</span> 
                            )}</h6>
                    <button 
                        className='btn cta error customerInfo-btn'
                        onClick={() => props.selectCustomer(null)}
                        >Close</button>
                </div>
            )
        }
    }

customerInfo.propTypes = {
    selectedCustomer: PropTypes.object,
    selectCustomer: PropTypes.func.isRequired
}

export default customerInfo