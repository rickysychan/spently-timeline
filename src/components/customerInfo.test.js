import React from 'react'
import {shallow} from 'enzyme'
import CustomerInfo from './CustomerInfo'
 
const sample = { 
        name: {first: "Eric", 
               last: 'hartman'}, 
        friends:['ioo', 'foo', 'moe'], 
        tags:['goo', 'moo', 'shoe'], 
        age: 29, 
        isActive: false, }

const mockSelectCustomer = jest.fn()
const customerInfo = shallow(<CustomerInfo 
                                selectedCustomer={sample} 
                                selectCustomer={mockSelectCustomer}/>)
const customerInfoEmpty = shallow(<CustomerInfo selectCustomer={mockSelectCustomer}/>)

describe('Client Info', () => {
    it('should render properly', () => {
        expect(customerInfo).toMatchSnapshot()
    })
    describe('when there is an object in selectedCustomer of `props`', () => {
        it('should render the customer info div', () => {
            expect(customerInfo.find('.customer-info').exists()).toBe(true)
        })
    })
    describe('when there isnt an object in selectedCustomer of `props`', () => {
        it('should render nothing', () => {
            expect(customerInfoEmpty.find('.customerInfo-no-customer').exists()).toBe(true)
        })
    })
    describe('when the user clicks the close button', () => {
        it('should invoke the selectCustomer method with null has parameter', ()=> {
            customerInfo.find('.customerInfo-btn').simulate('click')
            expect(mockSelectCustomer).toHaveBeenCalledWith(null)
        })
    })
})