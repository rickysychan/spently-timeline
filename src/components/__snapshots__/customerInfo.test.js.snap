// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Info should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <customerInfo
    selectCustomer={[Function]}
    selectedCustomer={
        Object {
            "age": 29,
            "friends": Array [
              "ioo",
              "foo",
              "moe",
            ],
            "isActive": false,
            "name": Object {
              "first": "Eric",
              "last": "hartman",
            },
            "tags": Array [
              "goo",
              "moo",
              "shoe",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="customer"
          className="customerInfo-image"
          src={undefined}
/>,
        <h6
          className="customerInfo-details"
>
          Full Name: 
          Eric hartman
</h6>,
        <h6
          className="customerInfo-details"
>
          Balance: 
</h6>,
        <h6
          className="customerInfo-details"
>
          Active: 
          false
</h6>,
        <h6
          className="customerInfo-details"
>
          Age: 
          29
</h6>,
        <h6
          className="customerInfo-details"
>
          About: 
</h6>,
        <h6
          className="customerInfo-details"
>
          Address: 
</h6>,
        <h6
          className="customerInfo-details"
>
          Friends: 
          <span>
                     
                    undefined, 
                     
          </span>
          <span>
                     
                    undefined, 
                     
          </span>
          <span>
                     
                    undefined, 
                     
          </span>
</h6>,
        <h6
          className="customerInfo-details"
>
          Tags: 
          <span>
                    goo, 
          </span>
          <span>
                    moo, 
          </span>
          <span>
                    shoe, 
          </span>
</h6>,
        <button
          className="btn cta error customerInfo-btn"
          onClick={[Function]}
>
          Close
</button>,
      ],
      "className": "customer-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "customer",
          "className": "customerInfo-image",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Full Name: ",
            "Eric hartman",
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Full Name: ",
          "Eric hartman",
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Balance: ",
            undefined,
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Balance: ",
          undefined,
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Active: ",
            "false",
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Active: ",
          "false",
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Age: ",
            29,
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Age: ",
          29,
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "About: ",
            undefined,
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "About: ",
          undefined,
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Address: ",
            undefined,
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Address: ",
          undefined,
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Friends: ",
            Array [
              <span>
                 
                undefined, 
                 
</span>,
              <span>
                 
                undefined, 
                 
</span>,
              <span>
                 
                undefined, 
                 
</span>,
            ],
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Friends: ",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "undefined, ",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "undefined, ",
              " ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "undefined, ",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "undefined, ",
              " ",
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "undefined, ",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "undefined, ",
              " ",
            ],
            "type": "span",
          },
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Tags: ",
            Array [
              <span>
                goo, 
</span>,
              <span>
                moo, 
</span>,
              <span>
                shoe, 
</span>,
            ],
          ],
          "className": "customerInfo-details",
        },
        "ref": null,
        "rendered": Array [
          "Tags: ",
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "goo, ",
            },
            "ref": null,
            "rendered": "goo, ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "moo, ",
            },
            "ref": null,
            "rendered": "moo, ",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "shoe, ",
            },
            "ref": null,
            "rendered": "shoe, ",
            "type": "span",
          },
        ],
        "type": "h6",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Close",
          "className": "btn cta error customerInfo-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Close",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="customer"
            className="customerInfo-image"
            src={undefined}
/>,
          <h6
            className="customerInfo-details"
>
            Full Name: 
            Eric hartman
</h6>,
          <h6
            className="customerInfo-details"
>
            Balance: 
</h6>,
          <h6
            className="customerInfo-details"
>
            Active: 
            false
</h6>,
          <h6
            className="customerInfo-details"
>
            Age: 
            29
</h6>,
          <h6
            className="customerInfo-details"
>
            About: 
</h6>,
          <h6
            className="customerInfo-details"
>
            Address: 
</h6>,
          <h6
            className="customerInfo-details"
>
            Friends: 
            <span>
                         
                        undefined, 
                         
            </span>
            <span>
                         
                        undefined, 
                         
            </span>
            <span>
                         
                        undefined, 
                         
            </span>
</h6>,
          <h6
            className="customerInfo-details"
>
            Tags: 
            <span>
                        goo, 
            </span>
            <span>
                        moo, 
            </span>
            <span>
                        shoe, 
            </span>
</h6>,
          <button
            className="btn cta error customerInfo-btn"
            onClick={[Function]}
>
            Close
</button>,
        ],
        "className": "customer-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "customer",
            "className": "customerInfo-image",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Full Name: ",
              "Eric hartman",
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Full Name: ",
            "Eric hartman",
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Balance: ",
              undefined,
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Balance: ",
            undefined,
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Active: ",
              "false",
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Active: ",
            "false",
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Age: ",
              29,
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Age: ",
            29,
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "About: ",
              undefined,
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "About: ",
            undefined,
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Address: ",
              undefined,
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Address: ",
            undefined,
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Friends: ",
              Array [
                <span>
                   
                  undefined, 
                   
</span>,
                <span>
                   
                  undefined, 
                   
</span>,
                <span>
                   
                  undefined, 
                   
</span>,
              ],
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Friends: ",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "undefined, ",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "undefined, ",
                " ",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "undefined, ",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "undefined, ",
                " ",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "undefined, ",
                  " ",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "undefined, ",
                " ",
              ],
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Tags: ",
              Array [
                <span>
                  goo, 
</span>,
                <span>
                  moo, 
</span>,
                <span>
                  shoe, 
</span>,
              ],
            ],
            "className": "customerInfo-details",
          },
          "ref": null,
          "rendered": Array [
            "Tags: ",
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "goo, ",
              },
              "ref": null,
              "rendered": "goo, ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "moo, ",
              },
              "ref": null,
              "rendered": "moo, ",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "shoe, ",
              },
              "ref": null,
              "rendered": "shoe, ",
              "type": "span",
            },
          ],
          "type": "h6",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Close",
            "className": "btn cta error customerInfo-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Close",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
